### PR TITLE
Fixed multiple issues within new gis result card

### PR DESCRIPTION
**Pull request by alexunique0519**
### Description
This PR contains multiple fixes for a series of tickets related to the new GIS result card. 

List of proposed changes:
- ADO-107419
- ADO-107420
- ADO-107431
- ADO-107513

### CODE REVIEW

File: components/ResultsPage/BenefitCard.tsx [hunk 1]
```diff
@@ -43,9 +43,12 @@ export const BenefitCard: React.VFC<{
...
     <div className="my-6 py-6 px-8 border border-[#6F6F6F] rounded">
-      <h2 id={benefitKey} className="h2 inline-flex justify-start items-center">
-        <span className="flex-1">{benefitName}</span>&nbsp;{eligibleFlag}
-      </h2>
+      <div className="inline">
+        <h2 id={benefitKey} className="inline align-sub h2">
+          {benefitName}
+        </h2>
+        {eligibleFlag}
+      </div>
```
Comment (MarcoGoC) [inline components/ResultsPage/BenefitCard.tsx:51]: this was the original code a few weeks ago, when was it changed? 
```
    <div className="my-6 py-6 px-8 border border-[#6F6F6F] rounded">
      <h2 id={benefitKey} className="h2">
        {benefitName}&nbsp;{eligibleFlag}
      </h2>
```

Comment (alexunique0519) [reply components/ResultsPage/BenefitCard.tsx:51]: Yesterday. Since the badge and the text are not vertically aligned.

Comment (MarcoGoC) [reply components/ResultsPage/BenefitCard.tsx:51]: what I mean the original code was ok.

Comment (alexunique0519) [reply components/ResultsPage/BenefitCard.tsx:51]: @MarcoGoC  You can switch back to it, and you will see that the badge is lower than the heading text, which is different from the mockup.

**Review by MarcoGoC — APPROVED** [review]

**Pull request merged**